File: dsa-problems/Probabilistic/solutions/cpp/PRB-015-median-uniforms-clt.cpp
#include <iostream>
#include <iomanip>
#include <vector>

using namespace std;

class Solution {
public:
    pair<double, double> medianClt(int n) {
        double mean = 0.5;
        double variance = 1.0 / (4.0 * n);
        return {mean, variance};
    }
};

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);

    int n;
    if (cin >> n) {
        Solution solution;
        auto res = solution.medianClt(n);
        cout << fixed << setprecision(6) << res.first << " " << res.second << "\n";
    }
    return 0;
}

File: dsa-problems/Probabilistic/solutions/java/PRB-015-median-uniforms-clt.java
import java.util.*;

class Solution {
    public double[] medianClt(int n) {
        double mean = 0.5;
        double variance = 1.0 / (4.0 * n);
        return new double[]{mean, variance};
    }
}

class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        if (sc.hasNextInt()) {
            int n = sc.nextInt();

            Solution solution = new Solution();
            double[] res = solution.medianClt(n);
            System.out.printf("%.6f %.6f\n", res[0], res[1]);
        }
        sc.close();
    }
}
